src/main/java/com/patreon/PatreonAPI.java

```diff
@@ -168,4 +212,44 @@ private String getVersion() throws IOException {
     prop.load(resourceAsStream);
     return prop.getProperty("version");
   }
+
+  /**
+   * Add fields[type]=fieldName,fieldName,fieldName as a query parameter to the request represented by builder
+   * @param builder A URIBuilder building a request to the API
+   * @param type A BaseResource annotated with {@link com.github.jasminb.jsonapi.annotations.Type}
+   * @param fields A list of fields to include.  Only fields in this list will be retrieved in the query
+   * @return builder
+   */
+  private URIBuilder addFieldsParam(URIBuilder builder, Class<? extends BaseResource> type, Collection<? extends Field> fields) {
+    List<String> fieldNames = new ArrayList<>();
+    for (Field f : fields) {
+      fieldNames.add(f.getPropertyName());
+    }
+    String typeStr = BaseResource.getType(type);
+    builder.addParameter("fields[" + typeStr + "]", join(fieldNames));
+
+    return builder;
+  }
+
+  /**
+   * Join a collection of strings with commas as separators.
+   */
+  private String join(Collection<? extends CharSequence> items) {
```

Comment:
Lil nit here, lets pull in my 1.8 PR, and use `String.join`

Reply:
👍